### PR TITLE
himalaya-vim: repo split & migrated

### PR DESCRIPTION
**Pull request by toastal**
Previously the Vim plugin was a part of the main repository, but the author has elected to split out the project and move away from GitHub to an open source Git forge @ https://git.sr.ht/~soywod/himalaya-vim/

As such, the previous plugin is currently broken, but now it can be updated indepentently.

###### Description of changes

@yanganto 

~~My config is currently broken because it's derived through home-manager, but the plugin errors in the exact same manner as the `himalaya` binary.~~ I know it says *not* to edit the generated Vim Nix file, but I kept getting 404s (with and without the new repository added... and this sort of thing happens to me a lot for not living in the West) so I took the path of least resistance.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by toastal** [comment]
Fixed my Himalaya config and this built and works!

**Comment by toastal** [comment]
While this waits in review, if anyone needs the copy/paste version for Vim plugin overrides:

```nix
  himalaya-vim = vimUtils.buildVimPluginFrom2Nix {
    pname = "himalaya-vim";
    version = "2022-10-09";
    src = fetchFromSourcehut {
      owner = "soywod";
      repo = "himalaya-vim";
      rev = "747ff2a6055839cb5b403b540883515af37b7117";
      sha256 = "sha256-0WP9JoKN/A5bmnRjQJnquk3mRC7NEflUAdqki7+zqBU=";
    };
    dependencies = with pkgs; [ himalaya ];
  };
```

I don't want continue force pushing to try to keep the branch up to date.

File: pkgs/applications/editors/vim/plugins/generated.nix [hunk 1]
```diff
@@ -3488,6 +3488,17 @@ final: prev:
...
   };
 
+  himalaya-vim = buildVimPluginFrom2Nix {
+    pname = "himalaya-vim";
+    version = "2022-10-10";
+    src = fetchgit {
+            url = "https://git.sr.ht/~soywod/himalaya-vim";
+            rev = "747ff2a6055839cb5b403b540883515af37b7117";
+            sha256 = "sha256-0WP9JoKN/A5bmnRjQJnquk3mRC7NEflUAdqki7+zqBU=";
+        };
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/vim/plugins/generated.nix:3498]:
```suggestion
    src = fetchgit {
      url = "https://git.sr.ht/~soywod/himalaya-vim";
      rev = "747ff2a6055839cb5b403b540883515af37b7117";
      sha256 = "sha256-0WP9JoKN/A5bmnRjQJnquk3mRC7NEflUAdqki7+zqBU=";
    };
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/editors/vim/plugins/generated.nix (hunk 1 above)
Comment (toastal) [reply pkgs/applications/editors/vim/plugins/generated.nix:3498]: @SuperSandro2000  This is a "bug" in the indentation of the Python script. You can check out other Vim plugins from SourceHut.

Comment (toastal) [reply pkgs/applications/editors/vim/plugins/generated.nix:3498]: https://github.com/NixOS/nixpkgs/blob/f0adf4fcae2aecdd6ade53cb5d4b94d3b7446efd/pkgs/applications/editors/vim/plugins/generated.nix#L3433-L3442

Here's an example of what I mean. Fixing this indentation bug is out of scope.

**Comment by divanorama** [comment]
fwiw, builds ok for me on {x86_64,aarch64}-{linux,darwin}

**Comment by toastal** [comment]
@SuperSandro2000 Should we address the Vim Python script indentation bug or not?

**Comment by figsoda** [comment]
I opened #201714 to fix the update script

**Comment by toastal** [comment]
@SuperSandro2000 I have rebased with https://github.com/NixOS/nixpkgs/pull/201714 which covers the indentation bug in Nixpkgs.

File: pkgs/applications/editors/vim/plugins/overrides.nix [hunk 2]
```diff
@@ -471,20 +471,9 @@ self: super: {
...
-    configurePhase = ''
-      cd vim
-      substituteInPlace plugin/himalaya.vim \
-        --replace 'if !executable("himalaya")' 'if v:false'
-    '';
-    postFixup = ''
-      mkdir -p $out/bin
-      ln -s ${himalaya}/bin/himalaya $out/bin/himalaya
-    '';
-  };
```
Comment (figsoda) [inline pkgs/applications/editors/vim/plugins/overrides.nix:487]: any reasons why those were removed?

Comment (toastal) [reply pkgs/applications/editors/vim/plugins/overrides.nix:487]: Whatever the rewrite was when the repository migrated to the open SourceHut, these were no longer required. I just rebuilt my Vim config to double check and it appears to work for me without the lines still.

File: pkgs/applications/editors/vim/plugins/overrides.nix [hunk 3]
```diff
@@ -471,20 +471,9 @@ self: super: {
     dependencies = with self; [ lush-nvim ];
   });
 
-  himalaya-vim = buildVimPluginFrom2Nix {
-    pname = "himalaya-vim";
-    inherit (himalaya) src version;
+  himalaya-vim = super.himalaya-vim.overrideAttrs (old: {
     dependencies = with self; [ himalaya ];
```
Comment (figsoda) [inline pkgs/applications/editors/vim/plugins/overrides.nix:475]: I tried it out and ran into this:
```
E605: Exception not caught: Himalaya CLI not found, see https://github.com/soywod/himalaya#installation
```
```suggestion
    postPatch = ''
      substituteInPlace plugin/himalaya.vim \
        --replace "if !executable('himalaya')" "if v:false"
      substituteInPlace autoload/himalaya/request.vim \
        --replace "'himalaya" "'${pkgs.himalaya}/bin/himalaya"
    '';
```

File: pkgs/applications/editors/vim/plugins/overrides.nix (hunk 2 above)
Comment (toastal) [reply pkgs/applications/editors/vim/plugins/overrides.nix]: I would bet I didn't have issues because I have the base Himalaya installed

**Review by figsoda — APPROVED** [review]

**Pull request merged**